File: src/main/java/bot/run/AbitVTBot.java
package bot.run;

import bot.commands.Command;
import bot.commands.Invoker;
import bot.commands.Receiver;
import bot.util.BotUpdatesHandler;
import bot.util.database.DatabaseConnection;
import bot.util.database.SqlFunctions;
import lombok.SneakyThrows;
import org.telegram.telegrambots.bots.TelegramLongPollingBot;
import org.telegram.telegrambots.meta.TelegramBotsApi;
import org.telegram.telegrambots.meta.api.objects.Update;
import org.telegram.telegrambots.meta.exceptions.TelegramApiException;
import org.telegram.telegrambots.updatesreceivers.DefaultBotSession;

import java.util.Map;

public class AbitVTBot extends TelegramLongPollingBot {

    //private Language language = Language.EN;
    private final BotUpdatesHandler updatesHandler;
    private final Invoker invoker;
    private final SqlFunctions sqlFunctions;

    //private final Translator translator = new Translator();

    public AbitVTBot() {
        this.updatesHandler = new BotUpdatesHandler(this);
        this.invoker = new Invoker(this);
        Receiver receiver = new Receiver(this);
        this.sqlFunctions = new SqlFunctions(new DatabaseConnection().getConnection());

        receiver.setCommands(invoker);
    }

    public static void main(String[] args) throws TelegramApiException {
        AbitVTBot bot = new AbitVTBot();
        TelegramBotsApi telegramBotsApi = new TelegramBotsApi(DefaultBotSession.class);
        telegramBotsApi.registerBot(bot);
    }

    @Override
    public String getBotUsername() {
        return System.getenv("BOT_NAME");
    }

    @Override
    public String getBotToken() {
        return System.getenv("BOT_TOKEN");
    }

    @SneakyThrows
    @Override
    public void onUpdateReceived(Update update) {
        updatesHandler.newUpdate(update);
    }

    public void invoke(String command, Update update) throws TelegramApiException {
        invoker.execute(command, update);
    }

    public Map<String, Command> availableCommands() {
        return invoker.getCommandMap();
    }

//    public Language getLanguage() {
//        return language;
//    }
//
//    public void setLanguage(Language language) {
//        this.language = language;
//    }
//
//    public Translator getTranslator() {
//        return translator;
//    }

    public SqlFunctions getSql(){
        return sqlFunctions;
    }


}

File: src/main/java/bot/run/AbitVTBot.kt
package bot.run

import bot.commands.Command
import bot.commands.Invoker
import bot.commands.Receiver
import bot.util.BotUpdatesHandler
import bot.util.CommandsUtil.notifyAdmin
import bot.util.Log
import bot.util.ResourceOperator
import bot.util.database.DatabaseConnection
import bot.util.database.SqlFunctions
import org.telegram.telegrambots.bots.TelegramLongPollingBot
import org.telegram.telegrambots.meta.TelegramBotsApi
import org.telegram.telegrambots.meta.api.objects.Update
import org.telegram.telegrambots.meta.exceptions.TelegramApiException
import org.telegram.telegrambots.updatesreceivers.DefaultBotSession

class AbitVTBot : TelegramLongPollingBot() {

    private val log = Log()

    private val updatesHandler: BotUpdatesHandler = BotUpdatesHandler(this)
    private val invoker: Invoker = Invoker(this)

    val sqlFunctions: SqlFunctions = SqlFunctions(DatabaseConnection().getConnection())
    val resourceOperator: ResourceOperator = ResourceOperator(this)

    init {
        Receiver(this).setCommands(invoker)
        log.turnOn()
        Runtime.getRuntime().addShutdownHook(Thread { log.turnOff() })
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val bot = AbitVTBot()
            val telegramBotsApi = TelegramBotsApi(DefaultBotSession::class.java)
            telegramBotsApi.registerBot(bot)
        }
    }

    override fun getBotToken(): String {
        return System.getenv("BOT_TOKEN")
    }

    override fun getBotUsername(): String {
        return System.getenv("BOT_NAME")
    }

    override fun onUpdateReceived(update: Update?) {
        try {
            updatesHandler.newUpdate(update!!)
        } catch (e: TelegramApiException) {
            log.telegramApiException(e, update!!)
        } catch (e: Exception) {
            log.appError(e, update!!)
            notifyAdmin(e, update)
        }
    }

    fun inv(command: String?, update: Update?) {
        invoker.execute(command, update!!)
    }

    fun availableCommands(): LinkedHashMap<String, Command> {
        return invoker.commandMap
    }
}